### ISSUE TITLE
Model evaluation issue

### ISSUE THREAD

**Issue by Le-Gris** [facebookresearch/jepa-intuitive-physics#4]
Interesting work, thank you for sharing the code! 

I've been interested in evaluating video models using your code, but I believe there may be an error in the `extract_losses` function from the `evaluation_code/evals/intuitive_physics/eval.py` file at line 459.

```python
preds = torch.zeros_like(preds,device=preds.device)
```

My understanding of the control flow there is that if the model is an MAE, then the code produces normalized (or not) targets, and then gets predictions from the encoder which are assigned to the `preds` variable. Weirdly enough, at line 459, the predictions are effectively zeroed out. I re-ran evaluations on VideoMAEv2-g without that line and I get substantial differences in accuracy on the IntPhys dataset, and minor differences on Inflevel. I haven't run the model on GRASP yet, but my guess is that results will also be different there. Currently, the code is effectively computing the L1 loss on the pixel values of the target images compared to a baseline of 0 for the MAE models. 

Am I reading this correctly? 

I've run into another issue with the following line of code in the same file and function as above:

```python
pieces = clip.unfold(2, frames_per_clip,stride).permute(0,2,-1,1,3,4).contiguous()
```

Whenever the `frame_steps` value in the config file is set to a higher value (say 15, or 20 as reported for the hyperparameter sweep in the paper), I get an error that dim 2 is smaller than `frames_per_clip` (set to 16) for certain videos (in particular, some gravity videos in Inflevel which happen to be a few seconds shorter than other videos in that dataset). It seems like some videos don't have enough total frames to produce even 1 window of 16 frames when the frame step is high. How did you deal with this problem?

A final related question: why is the `num_frames` value hardcoded to 99//frame_step in the same file and function as above? 

```python
sampling_rate,num_frames = frame_step ,99//frame_step
```

**Comment by garridoq** [comment]
> Currently, the code is effectively computing the L1 loss on the pixel values of the target images compared to a baseline of 0 for the MAE models.

Thanks for flagging this issue in the code. I believe this to be a remnant of some previous tinkering that was not cleaned up properly. I believe that the evaluations were ran before this experiment, but I'll rerun everything to check.
I'll come back with updated numbers.

> Whenever the frame_steps value in the config file is set to a higher value (say 15, or 20 as reported for the hyperparameter sweep in the paper), I get an error that dim 2 is smaller than frames_per_clip

This can be an issue on IntPhys especially (with ~100 frames videos), as the frame skip leads to a video size smaller than the context size. In that case we just ignored too high frameskips.

> A final related question: why is the num_frames value hardcoded to 99//frame_step in the same file and function as above?

This parameter is ignored and it's just a remnant of a previous version of the code. I'll look into all that you mentioned and clean up accordingly.

**Comment by garridoq** [comment]
I can confirm that our evaluations of videomaev2 were done before the "prediction zeroing out" mishap and that I was able to reproduce all of the numbers presented in the paper with the fixed code.
The only results that may have been impacted are the evaluation on IntPhys's private test set.  I will be rerunning them shortly and include updated numbers if there is a difference.

Could you elaborate on the setting where you see differences and the performance that you obtain ? 

It is possible that this is a model checkpoint difference, as I see that the models have been migrated to huggingface recently, whereas we used a version obtained by the previous download links, a while ago.

**Comment by Le-Gris** [comment]
Thanks for all the clarifications! I am indeed using the more recent videomaev2 model from huggingface. Here is my config file for reference (on IntPhys):

```
pretrain:
  enc_checkpoint_key: model
  model_name: pretrain_videomae_giant_patch14_224
  patch_size: 14
  folder: ../models/videomaev2/mae-g/
  checkpoint: vit_g_hybrid_pt_1200e.pth
  write_tag: videomaev2-g
  frames_per_clip: 16
  is_mae: true
  mae_decoder_blocks: -1
data:
  batch_size: 1
  resolution: 224
  stride_sliding_window: 2
  use_bfloat16: true
  frames_per_clip: 16
  context_lengths: [2, 4, 6, 8, 10, 12]
  frame_steps: [2, 5] # 10 actually doesn't work for intphys -- raises error!
tag: default
nodes: 1
tasks_per_node: 8
eval_name: intuitive_physics
dataset: intphys
```

As you can see, I'm using the pretrained videomaev2-g model. I guess one other difference with your reported hyperparameter sweep is the additional context length of 12. 

On each subset of the data I get a slightly higher result:

**reported**

```
O1: 55.0
O2: 65.0
O3: 58.33333
```

**mine**
```
O1: 56.66667
O2: 66.66667
O3: 68.33334
```

EDIT: I forgot to mention that these numbers are the maximum Relative Accuracy (avg) metric. I also compared the numbers for identical hyperparameters to make sure it's not just differences in my choice of hyperparameters, and it seems there are minor differences here and there.
